stages(kickstart): ensure test inputs pass schema validation

mvo5 opened the pull request:
Now that inputs can be relatively easily validated against the schema this should also be used for all the "good" test inputs to ensure that all tests test against valid inputs.

supakeen commented:
I'll take a look at this once #1438 merges and this gets rebased, it *feels* to me that we'd be checking the test inputs being valid twice so perhaps a single test to validate all test inputs and then not validating the test inputs in the other two test cases (the ones from #1438) would suit better? Unsure.

mvo5 commented:
> I'll take a look at this once #1438 merges and this gets rebased, it feels to me that we'd be checking the test inputs being valid twice so perhaps a single test to validate all test inputs and then not validating the test inputs in the other two test cases (the ones from #1438) would suit better? Unsure.

Thanks for this suggestion! I changed the code now to have a separate test that just double checks that all our test inputs are valid (I hope I did not misunderstood you here?). I also changed the negative schema test to not mix positive/negative schema validation (as the positive validation is already part of the TEST_CASES). Make the PR a bit more churn-ish but feels cleaner this way.

supakeen reviewed (approved):
Yea, it's a bit more boilerplate but it does feel nicer to separate out the concerns so the other test cases can assume that the input test data is valid.

This is also all 'figuring things out', so let's just do what feels right until we have more tests and find out what fits our actual requirements the best :)
